src/object/cli_ec.c

```diff
@@ -2250,6 +2250,7 @@ obj_ec_recov_task_init(struct obj_reasb_req *reasb_req, daos_obj_id_t oid,
 	D_ALLOC_ARRAY(fail_info->efi_stripe_sgls, iod_nr);
 	if (fail_info->efi_stripe_sgls == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
+	fail_info->efi_stripe_sgls_nr = iod_nr;
```

Comment:
efi_stripe_sgls_nr should be same as efi_nrecx_lists?
but not bad for this change.